### DISCUSSION TITLE
Directus Sdk.. specifying values for the fields property is not easy

### DISCUSSION THREAD

**Discussion by cloudcompute** [directus/directus#12912] [Feature Requests]
Pl. consider this snippet.

```
const response = await directus.items('pages').readByQuery({
    fields: ['page_name', 'contents.item:contents.title', 
    'contents.item:contents.description', 'contents.item: contents.filename_download'],

   filter: {
     // some filter
   },
}
```

In the ```fields``` value, the word ```contents``` (my collection) is repetitive while ```item``` is a ```directus``` keyword. It took me few hours to understand how to specify the values for the ```fields``` that I need.

Consider if I have a deeply nested collection and I need to fetch, let's say, 20 values. A lot of repetitive code would be there. Second point, based on my personal experience,  day before yesterday, I had changed the db schema i.e. collections and their fields three times. For every change that I did in the backend, it was not easy to reflect that change in the ```fields``` values. 

Assume I wrote the code and let's say, after six months, another developer is doing its maintenance. It won't be easy for him to figure out how things are working.

So may I suggest to use some standard specification for it.. use the industry-standard JSON format.

**Comment by rijkvanzanten** [comment]
> while item is a directus keyword

It's not. It's the name of the field in your schema 👍🏻 

> So may I suggest to use some standard specification for it.. use the industry-standard JSON format.

Industry-standard JSON format for defining fields? What standard would that be?

**Comment by cloudcompute** [comment]
> It's the name of the field in your schema

I have just two collections and 'item' is nowhere. Content collection schema has these fields: id, title, description, name.

Taken the following content from this [link](https://docs.directus.io/reference/query/)
> Many-To-Any (Union Types)
> Lets say we have a collection pages with a many-to-any field called sections that points to headings, paragraphs, and videos. We only want to fetch title and level from headings, body from paragraphs and source from videos. We can achieve that by using:
> sections.item:headings.title
> sections.item:headings.level
> sections.item:paragraphs.body
> sections.item:videos.source

Using this code as a reference, I specified the "fields" value as follows:

```
fields: ['page_name', 'contents.item:contents.title', 'contents.item:contents.description', 
'contents.item: contents.filename_download'],
```

Here is an example for specifying the values for the fields property. Technically, JSON is not a correct word, I am sorry. What I meant was the ability to define a hierarchical structure just like we do in Graphql's ```gql``` 

```
{
  cart {
    items_count
    name
    items {
      id
      qty
      product {
        name
        manufacturer {
          brand
        }
      }
    }
  }
}
```

**Reply by rijkvanzanten** [reply]
> I have just two collections and 'item' is nowhere.

Maybe you're not using a Many-to-Any relationship then, in which case the example you provided doesn't apply to your schema 🙂 

> What I meant was the ability to define a hierarchical structure just like we do in Graphql's gql

Ah in that case, just use GraphQL! 😄

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
